### DISCUSSION TITLE
Question/Bug? - Setting post excerpt

### DISCUSSION THREAD

**Discussion by Shu-Wan** [mmistakes/minimal-mistakes#2906] [Q&A] [answered]
- [x] This is a question about using the theme.
- [x] I have updated all gems with `bundle update`.
- [x] I believe this to be a bug with the theme --- not Jekyll, GitHub Pages or one of the bundled plugins.
- [x] I have tested locally with `bundle exec jekyll build`.

---

## Environment informations

- **Minimal Mistakes version:*4.1.0* 
- **`github-pages` or `jekyll` gem version:*3.3.1* 
- **Operating system: *__macOS__ Sierra 10.12.2* 

---

## Expected behavior

I wanted to set my own `excerpt` for posts instead an auto-generated one. Therefore, I added an `excerpt` value into my YAML front matter, but it doesn't work. The pages still use auto-generated one instead I added. 
## Steps to reproduce the behavior

I checked out the layout file, and find out `excerpt` is referenced in the `includes/archive_single.html` by the following code.

``` html
{% if post.excerpt %}<p class="archive__item-excerpt" itemprop="description">{{ post.excerpt | markdownify | strip_html | truncate: 160 }}</p>{% endif %}
```

I notice `itemdrop='description'` and `post.excerpt` are sourcing different content from the post. Therefore, I tried to change either of them to make them both source from `description` or `excerpt`. Then I can have my own `excerpt` rather than an auto-generated one.

For example
```html
{% if post.excerpt %}<p class="archive__item-excerpt" itemprop="description">{{ post.description | markdownify | strip_html | truncate: 160 }}</p>{% endif %}
```

Is it an inconsistency bug or I understand it incorrectly? All files are in my GitHub repo.

Thanks for the help!

**Comment by mmistakes** [comment, marked as answer]
Strange. It appears to be working correctly on the demo site.

If you check out [this archive page](https://mmistakes.github.io/minimal-mistakes/year-archive/)
and scroll down, the post titled "Layout: Excerpt (Defined)" has an excerpt that has been manually set.

![screen shot 2016-12-19 at 10 23 08 am](https://cloud.githubusercontent.com/assets/1376749/21318558/b372230e-c5d7-11e6-97e0-818dc49f8b33.png)

The [excerpt override](https://github.com/mmistakes/minimal-mistakes/blob/master/_includes/archive-single.html) was set just as you did with `excerpt: ` in the post's YAML Front Matter.

```
---
title: "Layout: Excerpt (Defined)"
excerpt: "This is a user-defined post excerpt. It should be displayed in place of the post content in archive-index pages."
categories:
  - Layout
  - Uncategorized
tags:
  - content
  - excerpt
  - layout
---

This is the post content. It should be displayed in place of the user-defined excerpt in archive-index pages.

This paragraph should be absent from an archive-index page where `post.excerpt` is shown.
```

Do you have a public repo somewhere and a link the source of the post that's not working? Not that it should matter but is it a post, a page, or a collection document?

**Comment by Shu-Wan** [comment]
Hi Michael, thanks for your reply!

I changed the `archive-single.html` to original and it works correctly this time.

I have no idea why it didn't work last night. Sorry for the trouble I made! 

Thanks again for your help and this amazing theme.

**Comment by bgub** [comment]
I found the issue for me was that I was running `bundle exec jekyll serve -i`. When I changed it to `bundle exec jekyll serve` it worked.

**Comment by will-7s** [comment]
Hello everyone,
I have a bug on my blog built on GitHub Minimal Mistakes.
Locally, when I run bundle exec jekyll serve, everything works well on localhost:4000 and the design is OK.
When I visit my blog online, I realised the design is not OK, and I don't know why. I need your help
Thanks in advance
